### ISSUE TITLE
add a warning on attributes that can be confused with global ones

### ISSUE THREAD

**Issue by tanriol** [rust-lang/rust#78519]
I was asked by a person new to Rust why `#[allow(unused_variables)]` does not work in the following code.
```rust
#[allow(dead_code)]
#[allow(unused_variables)]

fn trip() {
}

fn main() {
 let a =1;
// println!("{}",a);
}
```

It would be nice to have a warning here that these attributes are for the next item, not for the whole file as one could think from their position in file. Probably a warning when there's a blank line between item body and its attributes would work.

**Comment by phansch** [comment]
I think Clippy's `empty_line_after_outer_attr` is what you're looking for? https://rust-lang.github.io/rust-clippy/master/index.html

**Comment by tanriol** [comment]
Looks right, yes. I'd argue that it's a good idea to have it in `rustc` because this looks like a mistake a new user can accidentally do and they likely would not know about `clippy` yet. However, the note about possible false positives is not exactly optimistic.

**Comment by tanriol** [comment]
And even in Clippy it's allow-by-default, so you cannot really use it when you're a newcomer and have no idea about this specific problem.

**Comment by RodBurman** [comment]
Using a default set up and version 1.83 gives:

```
% cargo build
   Compiling warnglobal v0.1.0 (/Users/rod/code/rust/triage/warnglobal)
warning: unused variable: `a`
 --> src/main.rs:8:6
  |
8 |  let a =1;
  |      ^ help: if this is intentional, prefix it with an underscore: `_a`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: `warnglobal` (bin "warnglobal") generated 1 warning
    Finished `dev` profile
```

and

```
cargo clippy
    Checking warnglobal v0.1.0 (/Users/rod/code/rust/triage/warnglobal)
warning: unused variable: `a`
 --> src/main.rs:8:6
  |
8 |  let a =1;
  |      ^ help: if this is intentional, prefix it with an underscore: `_a`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: empty line after outer attribute
 --> src/main.rs:2:1
  |
2 | / #[allow(unused_variables)]
3 | |
  | |_
4 |   fn trip() {
  |   --------- the attribute applies to this function
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#empty_line_after_outer_attr
  = note: `#[warn(clippy::empty_line_after_outer_attr)]` on by default
  = help: if the empty line is unintentional remove it
help: if the attribute should apply to the crate use an inner attribute
  |
1 ~ #![allow(dead_code)]
2 ~ #![allow(unused_variables)]
  |

warning: `warnglobal` (bin "warnglobal") generated 2 warnings
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
```

So the compiler still produces a minimal warning , but clippy by default gives the full explanation.
I might suggest the compiler if it emits any warnings should suggest running clippy? Does anyone think that is a good idea?
